app/models/restart_signal_handler.rb

```diff
@@ -1,11 +1,18 @@
 # JobQueue locks a mutex, hence the need for a separate SignalHandler thread
 # Self-pipe is also best practice, since signal handlers can themselves be interrupted
-class SignalHandler
+class RestartSignalHandler
+  class << self
+    alias listen new
```

Comment:
because calling RestartSignalHandler.new is not very intention conveying :D 
alternatives:
 - make it all static methods ?
 - make listen do a new and then call run ... but that means run will be public or I'll have to use send